### ISSUE TITLE
Optimization algorithms ignore MaxEvaluationNumber parameter.

### ISSUE THREAD

**Issue by dmitrygrigoryev** [openturns/openturns#1054]
Hi!
I try to solve reliability analysis problem using FORM with the Cobyla optimization algorithm.  I've set the MaximumEvaluationsNumber to 2, but I'd get 5 calls of the PythonFunction. 

Here is my test code:
```python
import openturns as ot
ot.RandomGenerator.SetSeed(100)
bb_distr = ot.Normal()
i = 0
def blackbox(a):
  global i
  i+=1
  x = bb_distr.getSample(1)
  return x[0]
distr = ot.Normal()
vector = ot.RandomVector(distr)
model = ot.PythonFunction(1, 1, blackbox)
G = ot.CompositeRandomVector(model, vector)
event = ot.Event(G, ot.Greater(), 0)
algo = ot.Cobyla()
algo.setVerbose(True)
algo.setMaximumAbsoluteError(1e-10)
algo.setMaximumEvaluationNumber(11)
a = ot.FORM(algo, event, distr.getMean())
a.run()
print("Total number of function calls = %s" % str(model.getEvaluationCallsNumber()))
print("i = %s" % i)

# Both output parameters are equal to 5
# 
# Python version
# '3.6.0 |Anaconda custom (64-bit)| (default, Dec 23 2016, 12:22:00) \n[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]'
#
# OpenTURNS version
# 1.12rc1```

**Comment by jschueller** [comment]
additional evaluations are needed to compute the lagrange multipliers

**Comment by jschueller** [comment]
But, when using `setMaximumEvaluationNumber(11)` it ends up with 25 calls, some evaluations are lost in the equality constraint (nearestpointproblem), this is fixed by #1055.
